integrationTests/chainSimulator/common.go

```diff
@@ -43,3 +98,103 @@ func GenerateTransaction(sender []byte, nonce uint64, receiver []byte, value *bi
 		Signature: []byte(mockTxSignature),
 	}
 }
+
+// SendTransaction will send a transaction and return the result
+func SendTransaction(
+	t *testing.T,
+	cs ChainSimulator,
+	sender []byte,
+	nonce *uint64,
+	receiver []byte,
+	value *big.Int,
+	data string,
+	gasLimit uint64,
+) *transaction.ApiTransactionResult {
+	tx := GenerateTransaction(sender, *nonce, receiver, value, data, gasLimit)
+	txResult, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(tx, maxNumOfBlocksToGenerateWhenExecutingTx)
+	*nonce++
+	require.Nil(t, err)
+	require.NotNil(t, txResult)
+	require.Equal(t, transaction.TxStatusSuccess, txResult.Status)
+	if txResult.Logs != nil && txResult.Logs.Events != nil && len(txResult.Logs.Events) > 0 {
+		require.NotEqual(t, signalError, txResult.Logs.Events[0].Identifier)
+	}
+
+	return txResult
+}
+
+// RequireAccountHasToken checks if the account has the amount of tokens (can also be zero)
+func RequireAccountHasToken(
+	t *testing.T,
+	cs ChainSimulator,
+	token string,
+	address string,
+	value *big.Int,
+) {
+	tokens, _, err := cs.GetNodeHandler(0).GetFacadeHandler().GetAllESDTTokens(address, dataApi.AccountQueryOptions{})
```

Comment:
This is wrong, it will fail for multiple shard tests.
Revert back to original code.
Also, delete that `if value.Cmp(big.NewInt(0)) == 0`, it is redundant

Reply:
I reverted the code to work with multiple shards, it was a mistake.
That condition is to check the account doesn't have that token (value i zero), it is needed for other tests (e.g. NFT/SFT execute then burn)

Reply:
Then shouldn't we actually check something like `GetNFTTokenIDsRegisteredByAddress` or anything similar to check the token actually exists? Otherwise, you check nothing for those cases